libs/common/build.gradle

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import org.opensearch.gradle.info.BuildParams
+
+apply plugin: 'opensearch.publish'
+
+archivesBaseName = 'opensearch-common'
+
+dependencies {
+  // This dependency is used only by :libs:core for null-checking interop with other tools
+  compileOnly "com.google.code.findbugs:jsr305:3.0.2"
+
+  testImplementation "com.carrotsearch.randomizedtesting:randomizedtesting-runner:${versions.randomizedrunner}"
+  testImplementation "junit:junit:${versions.junit}"
+  testImplementation "org.hamcrest:hamcrest:${versions.hamcrest}"
+
+  testImplementation(project(":test:framework")) {
```

Comment:
The `:test:framework` brings whole bucket of other dependencies (even for tests only), why do we need it?

Reply:
Because the tests that have been refactored from core extend `OpenSearchTestCase` which needs to test framework. These are only used in the verification tests and aren't packaged in the distribution jar